tests/python/unittest/test_operator.py

```diff
@@ -2905,37 +2905,46 @@ def test_roipooling():
                            numeric_eps=1e-4, rtol=1e-1, atol=1E-4)
 
 
-def check_pad_with_shape(shape, xpu, pad_width, mode):
+def check_pad_with_shape(shape, xpu, pad_width, mode, dtype="float64"):
     # bind with label
-    X = mx.symbol.Variable('X')
+    real_types = ["float16", "float32", "float64"]
+    X = mx.symbol.Variable('X', dtype=dtype)
     Y = mx.symbol.Pad(data=X, mode=mode, pad_width=pad_width)
-    x = mx.random.uniform(-1, 1, shape, ctx=mx.cpu()).copyto(xpu)
+    if dtype in real_types:
```

Comment:
I think you do not really need this if-else branch at this moment as the caller of this function only passes real types.

Reply:
Removed. I had kept it there as I thought that it will be useful when integer support is added. But removed for now.